kronolith/lib/Kronolith.php

```diff
@@ -1909,7 +1909,14 @@ public static function sendITipNotifications(
         $view->event = $event;
         $view->imageId = $image->getContentId();
 
-        foreach ($event->attendees as $email => $status) {
+        /* Enhancement #3965: Only send updates to organizer if the user is not the organizer */
+        $receivers = $event->attendees;
+        if ($event->organizer && !$this->isUserEmail($event->creator, $event->organizer) ) {
+            /* TODO: Guess organizer name and status if he is one of the attendees */
+            $receivers = array($event->organizer => array('name' => $event->organizer, 'response' => self::RESPONSE_ACCEPTED));
```

Comment:
Quoting Ralf Lang <notifications@github.com>:

>> @@ -1909,7 +1909,14 @@ public static function sendITipNotifications(
>>          $view->event = $event;
>>          $view->imageId = $image->getContentId();
>>
>> -        foreach ($event->attendees as $email => $status) {
>> +        /* Enhancement #3965: Only send updates to organizer if  
>> the user is not the organizer */
>> +        $receivers = $event->attendees;
>> +        if ($event->organizer &&  
>> !$this->isUserEmail($event->creator, $event->organizer) ) {
>> +            /* TODO: Guess organizer name and status if he is one  
>> of the attendees */
>> +            $receivers = array($event->organizer => array('name'  
>> => $event->organizer, 'response' => self::RESPONSE_ACCEPTED));
>
> Right. Changing Attendance is not an "update" but a "response". It  
> needs its own constant and case in sendItipNotification.
>
> Kronolith has no UI for changing your attendance status and sending  
> an Itip to the organizer. (While IMP allows multiple answers to the  
> same Itip mail). Kronolith only allows to update the EVENT status.  
> https://www.ietf.org/rfc/rfc2446.txt 2.1.1 indicates that the event  
> status may only be edited by the organizer.

Probably one reason why some calendar apps will disallow changing the  
event if you are not the ORGANIZER.


> I could ** replace** or **duplicate** the "Status" dropdown of the  
> edit event redbox. There is still room in the row of  
> kronolithEventStatus, kronolithEventLocation. Don't know which  
> option is best.

I would say make the status drop down read-only and add a new field  
for response when the current user is not the ORGANIZER.

-- 
mike
The Horde Project
http://www.horde.org
https://www.facebook.com/hordeproject
https://www.twitter.com/hordeproject